Naughts_and_Crosses.py

```diff
@@ -1,122 +1,237 @@
+                self.board.empty_spaces.remove(choices)
+
+            else:
+                print('That space is not available douche. Choose again')
+                self.turn()
         else:
-            board_state[x_row][y_column] = self.symbol
-            return board_state
-
-class AI():
-	def __init__(self, symbol):
-		self.symbol = 'O'
-    def turn(self):
-        x_row = randrange(3)
-        y_column = randrange(3)
-        if board_state[x_row][y_column] == 'X' or board_state[x_row][y_column] == 'O':
-            turn(self)
+            choices = self.current_player.ai_choice()
+            if choices in self.board.empty_spaces:
+                self.board.board_state[choices[0]][choices[1]] = self.current_player.symbol
+                self.board.empty_spaces.remove(choices)
+            else:
+                self.turn()
+
+    def change_player(self): #Checks which players turn it is and then switches it then tells the next person who's go
+        if self.current_player == self.p1: # it is
+            self.current_player = self.p2
         else:
-            board_state[x_row][y_column] = self.symbol
-            return board_state
-
-
-
-  # Start the game here with asking the name and then getting into the start game function #
-  #Commit the Player1 turn function #
-  #Check win condition #
-  #Have AI or Player2 do their turn #
-  #Check win condition #
-  #Repeat until win condition is in there or board is full #
-  #Ask if they would like to play again, then ask if they would like to player against a second player or AI again #
-  
-
-
-
-
+            self.current_player = self.p1
+        print('It is now ' + self.current_player.name + "'s go")
+
+    def check_draw(self): #Checks the empty spaces list and if there is no empty spaces, it's a draw
+        if not self.board.check_free_spaces():
+            self.running = False
+            print("Well great. That's a draw.")
+            return True
+
+    def check_win(self): #Checks the board to see if there is a player symbol in all the spaces of each win condition
+        for i in range((self.board.board_size * 2) + 2):
+                if all(
+                    self.board.board_state[self.board.win_condition[i][j][0]][self.board.win_condition[i][j][1]] 
+                    == self.current_player.symbol for j in range(self.board.board_size)
+                    ):
+                    print(self.current_player.name + " wins! Well done")
+                    self.running = False
+                    return True
+
+    def end_game(self):  #Tells then to bugger off
+        print("Game's over, go home now... Unless...")
+
+    def _player1_assignment(self): #Asks the user to say if player 1 is a human or a robot
+        self.player1_type = (input("Is player 1 a human or a robot?  (h for Human or a for Ai)"  )).lower()
+        if self.player1_type != 'h' and self.player1_type != 'a':
```

Comment:
`is not in ['h', 'a']`

Or if you properly define these as player_type variables, then in those.